conjure-java-jackson-serialization/src/test/java/com/palantir/conjure/java/serialization/ObjectMappersTest.java

```diff
@@ -421,6 +423,39 @@ private void testTypeFactory(ObjectMapper mapper) {
         assertThat(mapper.getTypeFactory()).isInstanceOf(CaffeineCachingTypeFactory.class);
     }
 
+    @Test
+    public void testTypeFactoryCacheMetrics() throws IOException {
+        TaggedMetricRegistry registry = SharedTaggedMetricRegistries.getSingleton();
+        JsonDatabindTypefactoryCacheMetrics metrics = JsonDatabindTypefactoryCacheMetrics.of(registry);
+        Meter hit = metrics.hit();
+        Meter miss = metrics.miss();
+
+        ObjectMapper mapper = ObjectMappers.newServerJsonMapper();
+        long hitBefore = hit.getCount();
+        long missBefore = miss.getCount();
+
+        mapper.writeValue(ByteStreams.nullOutputStream(), new SimpleSerializable());
+
+        long hits = hit.getCount() - hitBefore;
+        long misses = miss.getCount() - missBefore;
+
+        assertThat(hits).isGreaterThan(0);
+        assertThat(misses).isOne();
+
+        // After writing the same type again, we should observe no additional misses.
+        missBefore = miss.getCount();
+        mapper.writeValue(ByteStreams.nullOutputStream(), new SimpleSerializable());
+        misses = miss.getCount() - missBefore;
+        assertThat(misses).isZero();
```

Comment:
should we check that hits > 0 here?

```suggestion
        assertThat(hit.getCount() - hits).isGreaterThan(0);
```